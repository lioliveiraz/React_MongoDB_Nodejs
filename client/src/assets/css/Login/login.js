import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme) => ({
  root: {
    height: "100vh",
    width: "100%",
    background: "#51608C",
  },
  main: {
    height: "75%",
    padding: "10px",
  },
  paper: {
    height: "100%",
    textAlign: "center",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    background: "#F1F1F1",
  },
  title: {
    fontSize: "20px",
    fontWeight: "bold",
  },
  image: {
    height: "15%",
  },
  form: {
    display: "flex",
    flexDirection: "column",
    height: "50%",
    margin: "5%",
  },
  input: {
    padding: "1rem",
  },
  button: {
    width: "50%",
    marginLeft: "auto",
    marginRight: "auto",
    marginTop: "5%",
  },
}));
