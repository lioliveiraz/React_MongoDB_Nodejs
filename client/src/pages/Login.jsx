import React, { useState, useEffect } from "react";
import Paper from "@material-ui/core/Paper";

import { connect } from "react-redux";
import { toast } from "react-toastify";
import { useStyles } from "../assets/css/Login/login";

import { handlerLogin } from "../store/actions/auth";
import loginValidator from "../helpers/validations/loginValidator";
import Input from "./../components/Base/Input";
import { Grid } from "@material-ui/core";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";

import padlock from "../assets/images/padlock.svg";

function Login({ handlerLogin, errors }) {
  const [userObject, setUserObject] = useState({});
  const [inputErrors, setInputErrors] = useState({});

  const classes = useStyles();

  const getUserInput = (value, name) => {
    setInputErrors({});
    setUserObject({
      ...userObject,
      [name]: value,
    });
  };

  const submit = async (e) => {
    e.preventDefault();
    const validationErrors = loginValidator(userObject);
    if (validationErrors) {
      setInputErrors(validationErrors);
    } else {
      handlerLogin(userObject);
    }
  };

  useEffect(() => {
    toast.error(errors);
  }, [errors]);

  return (
    <Grid
      container
      component="main"
      justify="center"
      alignItems="center"
      className={classes.root}
    >
      {" "}
      <Grid item xs={10} sm={8} md={4} className={classes.main}>
        <Paper elevation={3} className={classes.paper}>
          <img src={padlock} alt="padlock" className={classes.image} />
          <Typography variant="h3" color="primary" className={classes.title}>
            LOGIN
          </Typography>

          <form className={classes.form} data-cy="login-form">
            <Grid item className={classes.input}>
              <Input
                type="email"
                name="email"
                placeholder="johndow@email.com"
                required={true}
                getUserInput={getUserInput}
                error={inputErrors["email"] && inputErrors["email"]}
              />
            </Grid>
            <Grid item className={classes.input}>
              <Input
                type="password"
                name="password"
                required={true}
                getUserInput={getUserInput}
                error={inputErrors["password"] && inputErrors["password"]}
              />
            </Grid>

            <Button
              onClick={submit}
              variant="contained"
              color="secondary"
              className={classes.button}
              data-cy="button-submit"
              disabled={Object.keys(userObject).length != 2}
            >
              LogIn
            </Button>
          </form>
        </Paper>
      </Grid>
    </Grid>
  );
}

const mapStateToProps = (state) => ({
  errors: state.auth.errors,
});

export default connect(mapStateToProps, { handlerLogin })(Login);
