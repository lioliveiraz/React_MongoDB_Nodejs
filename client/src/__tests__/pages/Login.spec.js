import React from "react";
import { mount, shallow, spyOn } from "enzyme";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import Login from "./../../pages/Login";
import { validateTruthiness } from "./../utils/testUtils";

let initialState = {
  auth: {
    errors: null,
  },
};

describe("Login", () => {
  let wrapper;
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const historyMock = { push: jest.fn() };

  beforeEach(() => {
    wrapper = mount(
      <Router>
        <Provider store={store}>
          <Login history={historyMock} />
        </Provider>
      </Router>
    );
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render correctly", () => {
    const loginForm = wrapper.find("[data-cy='login-form']");
    const InputPassword = wrapper.find("[type='password']");
    const InputEmail = wrapper.find("[type='email']");
    const button = wrapper.find('[data-cy="button-submit"]');
    const img = wrapper.find("img");
    const pTag = wrapper.find("p");

    const elementsArr = [
      loginForm,
      InputEmail,
      button,
      InputPassword,
      img,
      pTag,
    ];
    elementsArr.forEach((el) => {
      validateTruthiness(el);
    });
  });

  it("Should  password error message display", async () => {
    const inputEmail = wrapper.find("input").at(0);
    const button = wrapper.find("button").at(0);

    await inputEmail.simulate("input", {
      target: { value: "email@email.com", name: "email" },
    });

    await button.simulate("click");

    expect(wrapper.text()).toContain('"password" is required');
  });

  it("Should error message display", async () => {
    const inputPassword = wrapper.find("input").at(1);
    const button = wrapper.find("button").at(0);

    await inputPassword.simulate("input", {
      target: { value: "12345678", name: "password" },
    });

    await button.simulate("click");

    expect(wrapper.text()).toContain('"email" is required');
  });
});
