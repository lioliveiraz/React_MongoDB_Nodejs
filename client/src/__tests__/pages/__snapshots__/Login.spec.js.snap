// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `ReactWrapper {}`;

exports[`Login should match snapshot 2`] = `"<main class=\\"MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center\\"> <div class=\\"MuiGrid-root makeStyles-main-2 MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-8 MuiGrid-grid-md-4\\"><div class=\\"MuiPaper-root makeStyles-paper-3 MuiPaper-elevation3 MuiPaper-rounded\\"><img src=\\"padlock.svg\\" alt=\\"padlock\\" class=\\"makeStyles-image-5\\"><p class=\\"MuiTypography-root makeStyles-title-4 MuiTypography-body1 MuiTypography-colorPrimary\\">LOGIN</p><form class=\\"makeStyles-form-6\\" data-cy=\\"login-form\\"><div class=\\"MuiGrid-root makeStyles-input-7 MuiGrid-item\\"><div class=\\"MuiFormControl-root MuiTextField-root makeStyles-root-9\\" data-cy=\\"input-email-wrapper\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"standard-secondary\\" id=\\"standard-secondary-label\\">Email<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"standard-secondary\\" name=\\"email\\" placeholder=\\"johndow@email.com\\" required=\\"\\" type=\\"email\\" data-cy=\\"input-email\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-12\\"><span>Email&nbsp;*</span></legend></fieldset></div></div></div><div class=\\"MuiGrid-root makeStyles-input-7 MuiGrid-item\\"><div class=\\"MuiFormControl-root MuiTextField-root makeStyles-root-9\\" data-cy=\\"input-password-wrapper\\"><label class=\\"MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"standard-secondary\\" id=\\"standard-secondary-label\\">Password<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"standard-secondary\\" name=\\"password\\" required=\\"\\" type=\\"password\\" data-cy=\\"input-password\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-12\\"><span>Password&nbsp;*</span></legend></fieldset></div></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedSecondary\\" tabindex=\\"0\\" type=\\"button\\" data-cy=\\"button-submit\\"><span class=\\"MuiButton-label\\">LogIn</span><span class=\\"MuiTouchRipple-root\\"></span></button></form></div></div></main>"`;
